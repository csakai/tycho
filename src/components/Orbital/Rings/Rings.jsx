import React from 'react';
import PropTypes from 'prop-types';
import { DoubleSide, Euler } from 'three';
import Scale from '../../../utils/Scale';
import MathService from '../../../services/MathService';
import TextureContainer from '../../../containers/TextureContainer';

export default class Rings extends React.Component {

    static propTypes = {
        outerRadius: PropTypes.number.isRequired,
        maps: PropTypes.array.isRequired,
        barycenterTilt: PropTypes.number.isRequired,
        scale: PropTypes.number
    }

    render() {
        const { outerRadius, scale, maps } = this.props;
        const tilt = MathService.toRadians(this.props.barycenterTilt);
        const size = Scale(outerRadius * 2, scale);
        const rotation = new Euler(tilt, 0, 0);

        return (
            <mesh rotation={rotation}>
                <planeGeometry
                    width={size}
                    height={size}
                />
                <TextureContainer
                    transparent={true}
                    side={DoubleSide}
                    textures={maps}
                />
            </mesh>
        );
    }
}
