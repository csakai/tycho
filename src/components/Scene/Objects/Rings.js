import THREE from 'three';
import Scale from 'engine/scale';

export default class Rings extends THREE.Mesh {
  constructor(data) {
    super();

    this.data = data;
    this.render();
  }

  render = () => {
    this.setGeometry();
    this.setMaterial();
  }

  setGeometry = () => {
    const size = Scale(this.data.outerRadius);
    this.geometry = new THREE.PlaneGeometry(size, size, 32);
  }

  setMaterial = () => {
    this.material = new THREE.MeshBasicMaterial({
      color: 0xffff00,
      side: THREE.DoubleSide
    });
  }
}
