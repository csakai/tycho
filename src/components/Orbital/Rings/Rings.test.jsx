import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Rings from './';

describe('Rings Component', () => {
    let component;

    beforeEach(() => {
        component = shallow(
            <Rings
                outerRadius={1200}
                barycenterTilt={20}
                maps={[]}
            />
        );
    });

    it('should render the rings successfully', () => {
        expect(toJson(component)).toMatchSnapshot();
    });
});
