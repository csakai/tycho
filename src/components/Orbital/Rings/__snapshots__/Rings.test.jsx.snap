// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rings Component should render the rings successfully 1`] = `
<mesh
  rotation={
    Euler {
      "_order": "XYZ",
      "_x": 0.3490658503988659,
      "_y": 0,
      "_z": 0,
    }
  }
>
  <planeGeometry
    height={0.0024}
    width={0.0024}
  />
  <TextureContainer
    side={2}
    textures={Array []}
    transparent={true}
  />
</mesh>
`;
